org.eclipse.jdt.ui.tests/ui/org/eclipse/jdt/ui/tests/quickfix/CleanUpTest.java

```diff
@@ -17450,29 +17452,28 @@ public void testOverriddenAssignment() throws Exception {
 				+ "public class E {\n" //
 				+ "    public boolean removeUselessInitialization() {\n" //
 				+ "        // Keep this comment\n" //
-				+ "        boolean reassignedVar;\n" //
-				+ "        reassignedVar = \"\\n\".equals(File.pathSeparator);\n" //
+				+ "        boolean reassignedVar = \"\\n\".equals(File.pathSeparator);\n" //
 				+ "        return reassignedVar;\n" //
 				+ "    }\n" //
 				+ "\n" //
 				+ "    public long removeInitForLong() {\n" //
 				+ "        // Keep this comment\n" //
 				+ "        long reassignedVar;\n" //
+				+ "        System.out.println();\n" //
 				+ "        reassignedVar = System.currentTimeMillis();\n" //
```

Comment:
Why doesn't the cleanup change this to:

  System.out.println();
  long reassignedVar = System.currentTimeMillis();

The declaration has no non-passive call so it can simply be removed and moved to the true initialization.

Reply:
I thought about that, but didn't for two reasons:

1. The cleanup does not do that now, even though it would make sense, as well
2. Some folks don't like their variable declarations interspersed with the rest of the code. I don't want to mess with those folks.

Reply:
I don't think option 2 qualifies as a reason for not doing a cleanup that was requested by the end-user.  If someone doesn't like a cleanup result, they can undo.  I think a reasonable solution would be to use the current behavior when there is no non-passive changes between declaration and initialization and do it the 2nd way when required.  Either way, you should not have a non-changing test in the middle without a name to suggest it is known to fail.

Reply:
You can't make assumptions about the end-user's preferences unless it is obviously ridiculous.  There are end-users who don't care about the declaration being moved and will wonder why you didn't do the cleanup as it is perfectly good Java syntax.  There are users who won't like the choice of a variable name we choose for a new for loop, etc., etc..  That's their issue, not ours.  It is a "not" an unusable scenario as you mention.  The key thing is: "does the cleanup break the code or change the behaviour".  If the answer is no, then you should probably do it.  Anyway, I'm not going to argue with you.  If you don't want to do it and want to make the cleanup only change the first declaration, fine, but change the test name so it is obvious that the test doesn't do anything and either add comments in the test or have the title describe why not.

Reply:
I made it a preference. Since it's on by default, I think the behaviour is 99% what you suggested.

Reply:
Thanks Thomas.  Regarding my earlier comments about a test that does not do the cleanup: a lot of cleanups break out these types of failure tests to a separate test that has multiple methods, etc.. that all have scenarios where the cleanup/quickfix is not applicable and usually have multiple methods with names to indicate why the cleanup fails (e.g. doNotChangeWhenVariablesUsedElsewhere()).  Then, the test simply verifies that no changes are made to the file so no "expected" string needed to test against.  It also is clear from the test that the cleanups don't apply in the scenarios.

Reply:
But that's not what's happening: the cleanup still happens, but in a different way, right? I can separate the "do the move" case into a different test method from the "don't do the move" case if that helps.

Reply:
No issue.  I was just clarifying my initial comment about the original test when there was no change being made.  I believe you had asked this at our meeting and so I clarified here.  Sorry for the confusion.